### PR TITLE
docs: mention hardenconfig in kconfig

### PR DESCRIPTION
**Pull request by laurenmurphyx64**
Adds a mention of the Kconfig hardening script hardenconfig to
the Kconfig documentation to promote awareness of its existence.

Signed-off-by: Lauren Murphy <lauren.murphy@intel.com>

### CODE REVIEW

File: doc/guides/build/kconfig/menuconfig.rst [hunk 1]
```diff
@@ -6,7 +6,10 @@ Interactive Kconfig interfaces
 There are two interactive configuration interfaces available for exploring the
 available Kconfig options and making temporary changes: ``menuconfig`` and
 ``guiconfig``. ``menuconfig`` is a curses-based interface that runs in the
-terminal, while ``guiconfig`` is a graphical configuration interface.
+terminal, while ``guiconfig`` is a graphical configuration interface. There is
```
Comment (nashif) [inline doc/guides/build/kconfig/menuconfig.rst:9]: I would put this in its own section, this is really not related to 'interactive Kconfig interfaces'.

**Review by nashif — CHANGES_REQUESTED** [review]

File: doc/guides/build/kconfig/menuconfig.rst (hunk 1 above)
Comment (laurenmurphyx64) [reply doc/guides/build/kconfig/menuconfig.rst:9]: That's true. We already have [a page under Security for the hardening tool](https://docs.zephyrproject.org/latest/security/hardening-tool.html), though - do you think we should move that documentation under Kconfig and mention it under Security, or leave it where it is and mention it in one of the other Kconfig pages (not Interactive Kconfig interfaces)? @ceolin

Comment (ceolin) [reply doc/guides/build/kconfig/menuconfig.rst:9]: Yep, the interactive section is not the right place. Though I think it is good link the hardening section in Kconfig section. May add a link to that in:
https://docs.zephyrproject.org/latest/guides/build/index.html#configuration-system-kconfig ?

Comment (laurenmurphyx64) [reply doc/guides/build/kconfig/menuconfig.rst]: Moved mention to main Kconfig page :+1:

**Review by nashif — APPROVED** [review]

**Review by aasthagrover91 — COMMENTED** [review]
@laurenmurphyx64  You might want to change the commit message since the mention is not related to interactive kconfig interfaces.

**Comment by laurenmurphyx64** [comment]
> @laurenmurphyx64 You might want to change the commit message since the mention is not related to interactive kconfig interfaces.

Changed, thanks for the catch! :+1:

**Review by aasthagrover91 — APPROVED** [review]
